### PR TITLE
feat(Date picker): Add localization support for date picker

### LINKED ISSUE #956: Need a localization support for datapicker
We need a localization  support for datepicker 
https://angular.carbondesignsystem.com/?path=/story/components-date-picker--single

We need support for the following languages `german` `finnish` `japanese` `mandarin` `spanish` `french` `italian` `korian` `portuguese`.

### PR DESCRIPTION
**Pull request by Donisius**
Closes #956

### CODE REVIEW

File: src/datepicker/datepicker.component.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ import rangePlugin from "flatpickr/dist/plugins/rangePlugin";
 import flatpickr from "flatpickr";
 import { NG_VALUE_ACCESSOR } from "@angular/forms";
 import { carbonFlatpickrMonthSelectPlugin } from "./carbon-flatpickr-month-select";
+import * as languages from "flatpickr/dist/l10n/index";
```
Comment (cal-smith) [inline src/datepicker/datepicker.component.ts]: Could we export this from the datepicker module, and have the input take the full object instead?

File: src/datepicker/datepicker.component.ts [hunk 2]
```diff
@@ -184,6 +193,8 @@ export class DatePicker implements OnDestroy, OnChanges, AfterViewChecked {
 	constructor(protected elementRef: ElementRef) { }
 
 	ngOnChanges(changes: SimpleChanges) {
+		// Global localization of flatpickr instances
+		flatpickr.localize(languages.default[this.language]);
```
Comment (cal-smith) [inline src/datepicker/datepicker.component.ts]: This changes the config globally ... which is probably what we want in most cases, but slightly inconsistent with how the options are presented (and doesn't allow for per-calendar language overrides) ... this should be something handled by the `flatpickrOptions` getter

Global translations are handled by the `i18n` service ... that's out of scope for this PR though, we can look at that later

File: src/datepicker/datepicker.component.ts (hunk 1 above)
Comment (Donisius) [reply src/datepicker/datepicker.component.ts]: When I try to export from the datepicker module I get the error:

Comment (Donisius) [reply src/datepicker/datepicker.component.ts]: I get the following error when I export FlatpickrLanguages from the datepicker module:
```
Type 'Record<key, CustomLocale>' is not assignable to type 'any[] | Type<any>'.
  Type 'Record<key, CustomLocale>' is missing the following properties from type 'Type<any>': apply, call, bind, prototype, and 5 more
```

**Comment by cyrusfurtado** [comment]
hi. Any idea when this would be merged.

File: src/datepicker/datepicker.component.ts (hunk 1 above)
Comment (cal-smith) [reply src/datepicker/datepicker.component.ts]: `export * as languages from "flatpickr/dist/l10n/index";` or even just leaving the import and doing `export languages;` causes that?

**Comment by cal-smith** [comment]
Cool! thanks for that @cyrusfurtado, there's definitely some useful features from #976 we should be able to incorporate

File: src/datepicker/datepicker.component.ts (hunk 2 above)
Comment (cyrusfurtado) [reply src/datepicker/datepicker.component.ts]: Changing the language config globally should be left as an option to be done in the DatePicker module imho as in my PR here https://github.com/IBM/carbon-components-angular/pull/976/files#diff-1e9013bc3a56c5ec0f998d5ec536e11cL19.

File: src/datepicker/datepicker.module.ts [hunk 3]
```diff
@@ -3,6 +3,9 @@ import { NgModule } from "@angular/core";
 import { CommonModule } from "@angular/common";
 import { DatePicker } from "./datepicker.component";
 
+import * as languages from "flatpickr/dist/l10n/index";
+export default languages;
```
Comment (cal-smith) [inline src/datepicker/datepicker.module.ts]: We don't want to default export `languages` ... does `export languages;` or `export { languages };` not work?

File: src/datepicker/datepicker.stories.ts [hunk 4]
```diff
@@ -150,6 +152,18 @@ storiesOf("Components|Date Picker", module)
 			</ibm-date-picker>
 		`,
 		props: {
+			language: select("Language of the calendar", [
```
Comment (zvonimirfras) [inline src/datepicker/datepicker.stories.ts]:
```suggestion
			language: select("Calendar language", [
```
? :)

File: src/datepicker/datepicker.stories.ts [hunk 5]
```diff
@@ -150,6 +152,18 @@ storiesOf("Components|Date Picker", module)
...
+				"en",
```
Comment (zvonimirfras) [inline src/datepicker/datepicker.stories.ts]: Is it useful to spread these over new lines?

File: src/datepicker/datepicker.module.ts (hunk 3 above)
Comment (Donisius) [reply src/datepicker/datepicker.module.ts]: They both give errors.

`export languages;` gives the following error:
```
Declaration or statement expected
```

`export { languages };` gives: 
```
...datepicker.module has no default export.
```

File: src/datepicker/datepicker.stories.ts (hunk 5 above)
Comment (Donisius) [reply src/datepicker/datepicker.stories.ts]: I suppose it looks better in one line 😄

File: src/datepicker/datepicker.module.ts (hunk 3 above)
Comment (cal-smith) [reply src/datepicker/datepicker.module.ts]: Hmm ... `export const languages = flatpickrLanguages;` perhaps? Weird that the object export wouldn't work

Comment (Donisius) [reply src/datepicker/datepicker.module.ts]: That gives me `Individual declarations in merged declaration 'languages' must be all exported or all local.`

Comment (cal-smith) [reply src/datepicker/datepicker.module.ts]: Okay now it's just getting silly

Comment (cal-smith) [reply src/datepicker/datepicker.module.ts]: Let's skip that for now ... ideally it should work with the i18n service down the road, but we do need _something_ working in the short term

Comment (cal-smith) [inline src/datepicker/datepicker.module.ts]: Don't default export the languages since we're not using them as inputs 👍 
```suggestion
```

**Comment by cal-smith** [comment]
Also there's some conflicts in the datepicker component ... let me know if you need help sorting those out

**Comment by cyrusfurtado** [comment]
Any idea when this could be available in master?

**Comment by cal-smith** [comment]
Just waiting on one more review then we can get it in! Do you have a product deadline we should know about?

**Comment by cyrusfurtado** [comment]
Is there a way to globally set the language for the `Datepicker` Module.

**Comment by cal-smith** [comment]
Not at the moment, though that's something we're still sorting out ... Ideally it should be set through the `i18n` module, but given how datepicker applies the locale I'm not sure if that's going to be possible ... Definitly open an issue for this enhancement :+1:

**Pull request merged**